driver/razeraccessory_driver.c

```diff
@@ -736,6 +738,50 @@ static ssize_t razer_attr_write_mode_breath(struct device *dev, struct device_at
     return count;
 }
 
+static ssize_t razer_attr_write_mode_starlight(struct device *dev, struct device_attribute *attr, const char *buf, size_t count)
+{
+    struct razer_accessory_device *device = dev_get_drvdata(dev);
+    struct razer_report report = {0};
+    unsigned char speed = 0;
+
+    if (count != 1 && count != 4 && count != 7) {
+        printk(KERN_WARNING "razeraccessory: Starlight accepts only 1, 4 or 7 bytes input (speed, [RGB], [RGB])");
+        return count;
+    }
+    speed = buf[0];
+
+    switch (device->usb_dev->descriptor.idProduct) {
+    case USB_DEVICE_ID_RAZER_KRAKEN_KITTY_EDITION:
+        switch(count) {
+        case 4: // Single colour mode
+            report = razer_chroma_extended_matrix_effect_starlight_single(VARSTORE, ZERO_LED, speed, (struct razer_rgb *)&buf[1]);
+            report.transaction_id.id = 0x1F;
+            break;
+
+        case 7: // Dual colour mode
+            report = razer_chroma_extended_matrix_effect_starlight_dual(VARSTORE, ZERO_LED, speed, (struct razer_rgb *)&buf[1], (struct razer_rgb *)&buf[4]);
+            report.transaction_id.id = 0x1F;
+            break;
+
+        default: // "Random" colour mode
+            report = razer_chroma_extended_matrix_effect_starlight_random(VARSTORE, ZERO_LED, speed);
+            report.transaction_id.id = 0x1F;
+            break;
+        }
+        break;
+
+    default:
+        printk(KERN_WARNING "razeraccessory: Unknown device\n");
```

Comment:
Maybe change to this so we can theoretically find the function easier in the future
```
printk(KERN_WARNING "razeraccessory %s: Unknown device\n", __func__);
```

Reply:
Wouldn't it make more sense to do this for all functions then? As of now, no other function does something like this even though there are quite a few devices in the `razeraccessory` driver which do not support all available effects.

Reply:
I guess? The logging in the drivers is really bad anyways so there's a lot that could be improved (not that it makes much of a difference given the general code quality in the driver imo). I'll just merge anyways, it's a minor detail and indeed `__func__` is used nowhere in the repo